libs/output-mapping/tests/Writer/Helper/SliceHelperTest.php

```diff
@@ -0,0 +1,349 @@
+        ];
+    }
+
+    /**
+     * @dataProvider sliceSourceWithSomeMappingOptionsIsNotSupportedProvider
+     */
+    public function testSliceSourceWithMappingHavingCsvOptionsIsNotSupported(
+        array $mapping,
+        string $expectedErrorMessage,
+    ): void {
+        $file = $this->temp->createFile('test.csv');
+        file_put_contents($file->getPathname(), '"id","name"');
+
+        $mappingSource = new MappingSource(
+            new LocalFileSource($file),
+        );
+        $mappingSource->setMapping($mapping);
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage($expectedErrorMessage);
+
+        SliceHelper::sliceFile($mappingSource);
+    }
+
+    public function sliceProvider(): Generator
+    {
+        yield 'file without manifest' => [
+            'originalMappingSource' => $this->createTestMappingSource('test1.csv', new Temp()),
+            'expectedData' => '"123","Test Name"',
+            'expectedManifestData' => [
+                'columns' => ['id', 'name'],
+            ],
+        ];
+        yield 'file with manifest' => [
+            'originalMappingSource' => $this->createTestMappingSourceHavingManifest('test2.csv', new Temp()),
+            'expectedData' => '"123";"Test Name"',
+            'expectedManifestData' => [
+                'delimiter' => ';',
+                'columns' => ['id', 'name'],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider sliceProvider
+     */
+    public function testSlice(
+        MappingSource $originalMappingSource,
+        string $expectedData,
+        array $expectedManifestData,
+    ): void {
+        /** @var LocalFileSource $originalSource */
+        $originalSource = $originalMappingSource->getSource();
+        $mappingSource = SliceHelper::sliceFile($originalMappingSource);
+
+        // manifest data
+        self::assertNotNull($mappingSource->getManifestFile());
+        $manifestFilePathName = $mappingSource->getManifestFile()->getPathname();
+        self::assertSame(
+            $originalSource->getFile()->getPathname() . '.manifest',
+            $manifestFilePathName,
+        );
+        self::assertManifestData($expectedManifestData, $manifestFilePathName);
+
+        // source
+        self::assertSource($originalSource, $mappingSource->getSource());
+
+        $dataFiles = FilesHelper::getDataFiles($originalSource->getFile()->getPath());
+        self::assertCount(1, $dataFiles);
+
+        /** @var FinderSplFileInfo $slicedDirectory */
+        $slicedDirectory = array_shift($dataFiles);
+
+        self::assertSame($originalSource->getFile()->getPathname(), $slicedDirectory->getPathname());
+        self::assertSlicedData($expectedData, $slicedDirectory->getPathname());
+    }
+
+    public function testSliceSources(): void
+    {
+        $originalMappingSource = $this->createTestMappingSource('test1.csv', $this->temp);
```

Comment:
úplně nerozumím tomu jakej je rozdíl mezi tímhle testem a tím předchozím s data itemem 'file with manifest'

Reply:
Jedno testuje jednu metodu, druhy testuje druhou. Jedno je slice podle parametru, druhy je slice pole MappingSource. V ramci  https://github.com/keboola/platform-libraries/pull/190 dost mozna necham jen jednu public metodu, uvidim.

Reply:
Aha, SliceHelper::sliceFile vs SliceHelper::sliceSources, no vzhledem k tomu, že druhý používá první, tak se to dost kryje